chill-scala/src/test/scala/com/twitter/chill/ClosureCleanerSpec.scala

```diff
@@ -67,5 +70,29 @@ class ClosureCleanerSpec extends WordSpec with Matchers {
       ClosureCleaner(fn)
       fn(10) should equal(420)
     }
+
+    "Handle nested closures in non-serializable object" in {
+      class NestedClosuresNotSerializable {
+        val irrelevantInt: Int = 1
+        def closure(name: String)(body: => Int => Int): Int => Int = body
+        def getMapFn: Int => Int = closure("one") {
+          def x = irrelevantInt
+          def y = 2
+          val fn = { a: Int => a + y }
+          fn
+        }
+      }
+
+      val fn = new NestedClosuresNotSerializable().getMapFn
+      ClosureCleaner(fn)
+      isSerializable(fn) shouldBe true
```

Comment:
is this false before? Can we add a test that checks that?

Ideally we see something fail until we run the cleaner.